solidity/contracts/BancorUpgradeConverter.sol

```diff
@@ -0,0 +1,236 @@
+        @param _formerVersions      true if the converter version is under 0.5
+    */
+    function copyConnectors(IBancorConverter _fromConverter, IBancorConverter _toConverter, bool _formerVersions)
+        private
+    {
+        uint256 virtualBalance;
+        uint32 weight;
+        bool isVirtualBalanceEnabled;
+        bool isPurchaseEnabled;
+        bool isSet;
+        for (uint16 i = 1; i < _fromConverter.convertibleTokenCount(); i++) {
+            address connectorAddress = _fromConverter.convertibleToken(i);
+            (virtualBalance, weight, isVirtualBalanceEnabled, isPurchaseEnabled, isSet) = readConnector(
+                _fromConverter,
+                connectorAddress,
+                _formerVersions
+            );
+            ISmartToken connectorToken = ISmartToken(connectorAddress);
+            _toConverter.addConnector(connectorToken, weight, isVirtualBalanceEnabled);
+        }
+    }
+
+    /**
+        @dev copies the conversion fee from the current converter to the new one
+
+        @param _fromConverter       current converter to copy from
+        @param _toConverter         new converter to add a conversion fee
+    */
+    function copyConvertionFee(IBancorConverter _fromConverter, IBancorConverter _toConverter)
+        private
+    {
+        uint32 conversionFee = _fromConverter.conversionFee();
+        _toConverter.setConversionFee(conversionFee);
+    }
+
+    /**
+        @dev copies the quick buy path array from the current converter to the new one
+
+        @param _fromConverter       current converter to copy from
+        @param _toConverter         new converter to add a path
+    */
+    function copyQuickBuyPath(IBancorConverter _fromConverter, IBancorConverter _toConverter)
+        private
+    {
+        uint256 quickBuyPathLength = _fromConverter.getQuickBuyPathLength();
+        if (quickBuyPathLength > 0) {
+            IERC20Token[] memory path = new IERC20Token[](quickBuyPathLength);
+            for (uint256 i = 0; i < quickBuyPathLength; i++) {
+                path[i] = _fromConverter.quickBuyPath(i);
+            }
+            _toConverter.setQuickBuyPath(path);
+        }
+    }
+
+    /**
+        @dev for each connector of the current converter, the function transfers its balance to the new converter.
+        notice that the function assumes that the new converter has exact the same connectors as the current
+        converter owned.
+
+        @param _fromConverter       current converter to withdraw from
+        @param _toConverter         new converter to deposit
+        @param _formerVersions      true if the converter version is under 0.5
+    */
+    function transferConnectorsBalances(IBancorConverter _fromConverter, IBancorConverter _toConverter, bool _formerVersions)
+        private
+    {
+        uint256 connectorBalance;
+        for (uint16 i = 1; i < _fromConverter.convertibleTokenCount(); i++) {
```

Comment:
again, this won't work for unlimited connectors, should document it and should use convertibleTokenCount instead

Reply:
changed to connectorTokenCount